hudi-client/src/test/java/org/apache/hudi/io/TestHoodieCommitArchiveLog.java

```diff
@@ -397,10 +402,81 @@ public void testArchiveCommitCompactionNoHole() throws IOException {
+    HoodieTestDataGenerator.createCommitFile(basePath, "105", dfs.getConf());
+    HoodieTestDataGenerator.createCommitFile(basePath, "106", dfs.getConf());
+    HoodieTestDataGenerator.createCommitFile(basePath, "107", dfs.getConf());
+
+    assertTrue(archiveLog.archiveIfRequired(jsc));
+
+    // read the file
+    Reader reader =
+        HoodieLogFormat.newReader(dfs, new HoodieLogFile(new Path(basePath + "/.hoodie/.commits_.archive.1_1-0-1")),
+            HoodieArchivedMetaEntry.getClassSchema());
+    List<IndexedRecord> readRecords = new ArrayList<>();
+    // read the avro blocks and validate the number of records written in each avro block
+    while (reader.hasNext()) {
+      HoodieAvroDataBlock blk = (HoodieAvroDataBlock) reader.next();
+      List<IndexedRecord> records = blk.getRecords();
+      readRecords.addAll(records);
+    }
+
+    HoodieArchivedMetaEntry metaEntry = (HoodieArchivedMetaEntry)SpecificData.get()
+        .deepCopy(HoodieArchivedMetaEntry.getClassSchema(), readRecords.get(0));
+
+    assertEquals("operationType must be UNKNOWN by default.",
+        metaEntry.getHoodieCommitMetadata().getOperationType(), "UNKNOWN");
+    reader.close();
+  }
+
   private void verifyInflightInstants(HoodieTableMetaClient metaClient, int expectedTotalInstants) {
     HoodieTimeline timeline = metaClient.getActiveTimeline().reload()
         .getTimelineOfActions(Sets.newHashSet(HoodieTimeline.CLEAN_ACTION)).filterInflights();
     assertEquals("Loaded inflight clean actions and the count should match", expectedTotalInstants,
         timeline.countInstants());
   }
+
+  @Test
+  public void testCommitMetadataConverter() {
+    HoodieCommitMetadata hoodieCommitMetadata = new HoodieCommitMetadata();
+    hoodieCommitMetadata.setOperationType(WriteOperationType.INSERT);
+
+    HoodieWriteConfig cfg = HoodieWriteConfig.newBuilder().withPath(basePath)
+        .withSchema(HoodieTestDataGenerator.TRIP_EXAMPLE_SCHEMA).withParallelism(2, 2).forTable("test-commitMetadata-converter")
+        .withCompactionConfig(HoodieCompactionConfig.newBuilder().retainCommits(1).archiveCommitsWith(2, 5).build())
+        .build();
+    metaClient = HoodieTableMetaClient.reload(metaClient);
+    HoodieCommitArchiveLog archiveLog = new HoodieCommitArchiveLog(cfg, metaClient);
+
+    Class<?> clazz  = HoodieCommitArchiveLog.class;
+    try {
+      Method commitMetadataConverter = clazz.getDeclaredMethod("commitMetadataConverter", HoodieCommitMetadata.class);
+      commitMetadataConverter.setAccessible(true);
+      org.apache.hudi.avro.model.HoodieCommitMetadata expectedCommitMetadata =
+          (org.apache.hudi.avro.model.HoodieCommitMetadata) commitMetadataConverter.invoke(archiveLog, hoodieCommitMetadata);
+      assertEquals(expectedCommitMetadata.getOperationType(), WriteOperationType.INSERT.toString());
+    } catch (NoSuchMethodException e) {
+      e.printStackTrace();
```

Comment:
Instead of e.printStackTrace() in all these catch blocks, can you throw these exception for the test to fail